crates/pathfinder/src/rpc.rs

```diff
@@ -243,15 +246,8 @@ Hint: If you are looking to run two instances of pathfinder, you must configure
                     .await
             },
         )?;
-        module.register_async_method("starknet_getNonce", |params, context| async move {
-            #[derive(Debug, Deserialize)]
-            struct NamedArgs {
-                contract_address: ContractAddress,
-            }
-            context
-                .get_nonce(params.parse::<NamedArgs>()?.contract_address)
-                .await
-        })?;
+        Self::register_method::<get_nonce::GetNonce>(&mut module)
+            .with_context(|| format!("Registering {}", get_nonce::GetNonce::NAME))?;
```

Comment:
I guess the context could be moved into `Self::register_method`, there's no reason for it not to return `anyhow::Error` and it'll shrink boilerplate even more :thinking: .